Sources/PersistentCollections/PersistentDictionary/PersistentDictionary+Filter.swift

```diff
@@ -17,14 +17,10 @@ extension PersistentDictionary {
     var result: PersistentDictionary = [:]
     for item in self {
       guard try isIncluded(item) else { continue }
-      // FIXME: We could recover the key's hash from self.
-      // (As many bits of it as we need to insert it into the new tree.)
-      // However, this requires a series of `UInt32._bit(ranked:)` invocations
-      // that may or may not be meaningfully better than simply rehashing the
-      // key.
+      // FIXME: We could do this as a structural transformation.
```

Comment:
👍

Reply:
Consider that structural formations must deal with compaction in order to guarantee a canonical representation, which slightly complicates the matter.